### ISSUE TITLE
Types are handled wrong when using namespaces

### ISSUE THREAD

**Issue by peckto** [Fraunhofer-AISEC/cpg#1126]
# Unexpected behavior

When using namespaces, types are not associated with the correct namespace.
As a result, function calls are not resolved.

# Code of interest

```c++
namespace ABC {
    namespace XYZ {
        struct A {
            A();
            void foo();
        };
    }
}

using namespace ABC::XYZ;
using namespace ABC;

int main() {
        ABC::XYZ::A a;
        XYZ::A b;
        A c;

        a.foo();
        b.foo();
        c.foo();

	return 0;
}
```

All three variables (`a`, `b`, `c`) should have the same type.
![image](https://user-images.githubusercontent.com/2060591/224950794-a245d84a-70f8-47a8-8d74-e156d8fa7faa.png)

Similar behavior:
```c++
namespace ABC {
    namespace XYZ {
        struct A {
            void foo();
        };
    }
}


namespace ABC {
    namespace XYZ {
        void foo() {
            ABC::XYZ::A b;
            XYZ::A a;
            A c;

           a.foo();
           b.foo();
           c.foo();
        }
    }
}
```

# Extensions to the library, e.g. added or changed LanguageFrontends or Passes

None

# Changes to the configuration

None

**Comment by oxisto** [comment]
With respect to the second example: Are you sure they all point to the same struct? Because when I try to compile this, I get the following linker error:
```
Undefined symbols for architecture arm64:
  "ABC::XYZ::A::foo()", referenced from:
      ABC::XYZ::foo() in spaces-3074e4.o
```

**Comment by oxisto** [comment]
I am beginning to hate C++. Why the heck is `XYZ::A` inside the name space `ABC::XYZ` pointing to `ABC::XYZ::A`...

**Comment by oxisto** [comment]
@peckto this might be solved with #1502
